inttest/common/util.go

```diff
@@ -190,3 +195,29 @@ func WaitForLease(ctx context.Context, kc *kubernetes.Clientset, name string, na
 		return *lease.Spec.HolderIdentity != "", nil
 	})
 }
+
+// VerifyKubeletMetrics checks whether we see container and image labels in kubelet metrics.
+// It does it via polling as it takes some time for kubelet to start reporting metrics.
+func VerifyKubeletMetrics(ctx context.Context, kc *kubernetes.Clientset, node string) error {
+
+	return Poll(ctx, func(ctx context.Context) (done bool, err error) {
+
+		metrics, err := kc.CoreV1().RESTClient().Get().AbsPath("/api/v1/nodes/worker0/proxy/metrics/cadvisor").Param("format", "text").DoRaw(ctx)
```

Comment:
This is always checking metrics from worker0, but is called from within a loop over all worker nodes. I suggest to introduce the node name as an extra function argument.

Reply:
do'h. that _was_ the intention but apparently the intention never leaved my head 😄  will fix

Reply:
done